adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java

```diff
@@ -0,0 +1,43 @@
+package de.adorsys.xs2a.adapter.adorsys;
+
+import de.adorsys.xs2a.adapter.api.RequestHeaders;
+import de.adorsys.xs2a.adapter.api.config.AdapterConfig;
+import de.adorsys.xs2a.adapter.api.http.Interceptor;
+import de.adorsys.xs2a.adapter.api.http.Request;
+import de.adorsys.xs2a.adapter.impl.http.RequestSigningInterceptor;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class AdorsysSigningHeadersInterceptor implements Interceptor {
+
+    private final RequestSigningInterceptor signingInterceptor;
+    private final boolean requestSigningEnabled =
+        Boolean.parseBoolean(AdapterConfig.readProperty("adorsys.request_signing.enabled", "false"));
+
+    public AdorsysSigningHeadersInterceptor(RequestSigningInterceptor signingInterceptor) {
```

Comment:
I propose to the Keystore as a parameter and hide the creation of RequestSigningInterceptor

Reply:
The tests should be adjusted then. Working on it.

Reply:
done